### PR TITLE
feat(styling): add styling methods

### LINKED ISSUE #20: Adding styling methods
### Description:
1. We can implement a utility method to combine styles, and a hook for exposing styles to components with style overrides.

### Outcome:
1. “sx” method will be implemented
2. A makeStyles() util which gives useStyles() hook will be implemented
3. Add a separate section for Utils in Storybook
4. Add stories for styling like 	stylx, makeStyles and useStyles.
    - Use the Email component to override style.

### PR DESCRIPTION
**Pull request by niloysikdar**
### This PR covers:
1. Add “sx” method
2. Implemented makeStyles() util which gives useStyles() hook
3. Add tests for `sx` and `makeStyles`
4. Add stories for `sx` and `makeStyles`

Capture12

resolves #20

### CODE REVIEW

File: src/utils/makeStyles.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  return target;
+};
+
+export const makeStyles = (classes: CSSClasses) => {
+  return (classes_?: CSSClasses) => {
+    return mergeStyles(classes, classes_ as CSSClasses);
+  };
+};
```
Comment (agentmilindu) [inline src/utils/makeStyles.ts:19]:
```suggestion
export const makeStyles = (classes: CSSClasses) => {
  return ({ classes: overrideClasses }: {classes?: CSSClasses}) => {
    return mergeStyles(classes, overrideClasses);
  };
};
```

**Review by agentmilindu — CHANGES_REQUESTED** [review]

File: src/utils/stylings.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const initialValue: CSSProperties = {};
+
+  const result = args.reduce((prev: CSSProperties, curr: CSSProperties) => {
+    return { ...prev, ...curr };
+  }, initialValue);
+
+  return result;
+};
+
+export const makeStyles = (classes: CSSClasses) => {
```
[the remark this answers is not in the document]
Comment (niloysikdar) [reply src/utils/stylings.ts]: Done

File: src/utils/makeStyles.test.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  hello: {
+    backgroundColor: 'red',
+    borderRadius: '5px',
+  },
+};
+
+describe('makeStyles tests', () => {
+  test('using makeStyles and useStyles with style1 and style2 should return expectedStyle', () => {
+    const useStyles = makeStyles(style1);
+    const style = useStyles(style2);
```
[the remark this answers is not in the document]
Comment (niloysikdar) [reply src/utils/makeStyles.test.ts]: Done

File: src/utils/makeStyles.ts (hunk 1 above)
Comment (niloysikdar) [reply src/utils/makeStyles.ts]: Done

**Pull request merged**